src/borg/archiver/extract_cmd.py

```diff
@@ -56,33 +56,51 @@ def do_extract(self, args, repository, manifest, archive):
         else:
             pi = None
 
-        for item in archive.iter_items(filter):
-            archive.preload_item_chunks(item, optimize_hardlinks=True)
+        for item in archive.iter_items():
             orig_path = item.path
-            if strip_components:
-                item.path = os.sep.join(orig_path.split(os.sep)[strip_components:])
-            if not args.dry_run:
-                while dirs and not item.path.startswith(dirs[-1].path):
-                    dir_item = dirs.pop(-1)
-                    try:
-                        archive.extract_item(dir_item, stdout=stdout)
-                    except BackupError as e:
-                        self.print_warning_instance(BackupWarning(remove_surrogates(dir_item.path), e))
+            components = orig_path.split(os.sep)
+            stripped_path = os.sep.join(components[strip_components:])
+
+            if not stripped_path:
+                continue
+            item.path = stripped_path
+
+            is_matched = matcher.match(orig_path)
+            log_prefix = "+" if is_matched else "-"
```

Comment:
if you don't use log_prefix except in line 72, move 69 directly above that line.